src/Host/Protocol/Test/UserProfileServicePipe/UserProfileServicePipeTest.cs

```diff
@@ -0,0 +1,140 @@
+                            f.ShouldNotThrow();
+                        } else {
+                            Func<Task> f = () => RUserProfileCreator.CreateProfileAsync(serverTimeOutms: serverTimeOut, clientTimeOutms: clientTimeOut, userProfileService: creator);
+                            await f.ShouldThrowAsync<Exception>();
+                        }
+                    }
+                } finally {
+                    testDone.Set();
+                }
+            }).DoNotWait();
+
+            using (CancellationTokenSource cts = new CancellationTokenSource(TimeSpan.FromMilliseconds(clientTimeOut))) {
+                UserProfileResultMock result = await CreateProfileClientTestWorkerAsync(input, cts.Token);
+                if (isFuzzTest) {
+                    // fuzz test parsing succeeded, the creator always fails for this test.
+                    result?.Error.Should().Be(13);
+                } else {
+                    if (isValidParse) {
+                        result.Error.Should().Be((uint)(isValidAccount ? 0 : 13));
+                        result.ProfileExists.Should().Be(isExistingAccount);
+                    } else {
+                        result.Should().BeNull();
+                    }
+                }
+            }
+
+            testDone.Wait();
+        }
+
+        [CompositeTest]
+        // valid test data
+        [InlineData("{\"Username\":\"testname\", \"Domain\":\"testdomain\", \"Password\":\"testPassword\"}", "testname", "testdomain", "testPassword", true, true, false)]
+        // Missing quote
+        [InlineData("{Username\":\"testname\", \"Domain\":\"testdomain\", \"Password\":\"testPassword\"}", null, null, null, false, false, false)]
+        // Missing closing parenthesis
+        [InlineData("{", null, null, null, false, false, false)]
+        // empty json object
+        [InlineData("{}", null, null, null, false, false, false)]
+        // No username, domain and password
+        [InlineData("{\"Username\":, \"Domain\":, \"Password\":}", null, null, null, false, false, false)]
+        // empty username domain password
+        [InlineData("{\"Username\": \"\", \"Domain\": \"\", \"Password\": \"\"}", "", "", "", true, false, false)]
+        // whitespace input string
+        [InlineData("                     ", null, null, null, true, false, false)]
+        // empty string
+        [InlineData("", null, null, null, false, false, false)]
+        public async Task CreateProfileTest(string input, string username, string domain, string password, bool isValidParse, bool isValidAccount, bool isExistingAccount) {
+            var creator = UserProfileCreatorMock.Create(username, domain, password, isValidParse, isValidAccount, isExistingAccount);
+            await CreateProfileTestRunnerAsync(creator, input, isValidParse, isValidAccount, isExistingAccount, 500, 500);
+        }
+
+        [Test]
+        [Category.FuzzTest]
```

Comment:
I suggested having a generic category for fuzz stuff, so that they could be excluded from test runs altogether (to get deterministic runs if we need to).